src/diffusers/pipelines/controlnet/pipeline_controlnet.py

```diff
@@ -837,6 +843,20 @@ def __call__(
                 second element is a list of `bool`s indicating whether the corresponding generated image contains
                 "not-safe-for-work" (nsfw) content.
         """
+        # We do this first to capture the "true" call values. If we do this at a later point in time,
+        # we cannot ensure that the true call values weren't changed during the process.
+        # We update the `generator` later, though as we define a new generator in case it was passed as `None`.
+        if generator is None:
+            seed = random.randint(0, MAX_SEED)
+            generator = torch.manual_seed(seed)
+
+        workflow = None
```

Comment:
Let's try to move everything under `return_workflow:`

Reply:
I need to understand this bit. Shouldn't we do the generator initialization first regardless of `return_workflow`? I think we should also do before `prepare_latents()` is called with `generator`, otherwise it becomes incorrect.

Reply:
I don't think we need to create or measure a generator if we don't save the workflow